src/lib/server/cart.ts

```diff
@@ -69,7 +69,8 @@ export async function addToCartInDb(
 	if (
 		params.user.userRoleId === POS_ROLE_ID
 			? !product.actionSettings?.retail?.canBeAddedToBasket
-			: !product.actionSettings?.eShop?.canBeAddedToBasket
+			: !product.actionSettings?.eShop?.canBeAddedToBasket &&
+			  !product.actionSettings?.nostr.canBeAddedToBasket
```

Comment:
The mode ('eShop', 'nostr', 'pos'...) should be added as a param to  `addToCartInDb` and the check should depend on that mode.

Because with this PR a user could still add a nostr-exclusive item using the API instead of using NostR